File: CSharpRewrite/CSharpRewrite/Assignments/K08/K08.cs
using CSharpRewrite.Shared;

namespace CSharpRewrite.Assignments.K08;

public class K08 : IQuestion
{
    public void Question1()
    {
        var numbers = new List<int>();
        try
        {
            for (var i = 0; i < 3; i++)
            {
                Console.Write($"{i + 1}つ目の値を入力してください。＞");
                numbers.Add(int.Parse(Console.ReadLine() ?? "0"));
            }

            Console.WriteLine("どちらを調べますか？");
            Console.Write("（０：最大値　１：最小値）＞");

            switch (int.Parse(Console.ReadLine() ?? "0"))
            {
                case 0:
                    Console.WriteLine($"{numbers.Count}の中で最大値は{numbers.Max()}です。");
                    break;
                case 1:
                    Console.WriteLine($"{numbers.Count}の中で最小値は{numbers.Min()}です。");
                    break;
            }
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question2()
    {
        Console.WriteLine("冒険が今始まる！");
        var rng = new Random();
        var playerHp = 200 + rng.Next(0, 101);
        Console.WriteLine(Adventure.GameLoop(playerHp));
    }

    public void Question3() { }

    public void Question4() { }
}

File: TypeScript/assignments/K08/K08.ts
import Executable from "../../shared/Executable.ts"
import { gameLoop } from "./Adventure.ts";
import { getRandomNumber } from "../../shared/Utility.ts";

export default class K08 extends Executable {
    question1(): void {
        const numbers = [0, 0, 0];
        for (let i = 0; i < 3; i++) {
            numbers[i] = Number.parseInt(prompt(`${i + 1}つ目の値を入力してください。＞`) ?? "0");
        }
        console.log("どちらを調べますか？");
        
        switch (Number.parseInt(prompt("（０：最大値　１：最小値）＞") ?? "0")) {
            case 0:
                console.log(`${numbers.length}つの中で最大値は${Math.max(...numbers)}です。`);
                break;
            case 1:
                console.log(`${numbers.length}つの中で最小値は${Math.min(...numbers)}です。`);
                break;
            default:
                break;
        }
    }

    question2(): void {
        console.log("冒険が今始まる！");
        const playerHp = 200 + getRandomNumber(101);
        console.log(gameLoop(playerHp));
    }

    question3(): void {
        return;
    }

    question4(): void {
        return;
    } 
}